src/object/cli_obj.c

```diff
@@ -1996,6 +2037,18 @@ obj_comp_cb(tse_task_t *task, void *data)
 		obj_retry_cb(task, obj, obj_auxi);
 
 	if (!obj_auxi->io_retry) {
+		if (obj_auxi->opc == DAOS_OBJ_RPC_SYNC &&
+		    task->dt_result != 0) {
+			struct daos_obj_sync_args	*sync_args;
+
+			sync_args = dc_task_get_args(task);
+			D_ASSERT(sync_args->epoch != NULL);
```

Comment:
seems that in extreme case, it possible failed for example -NOMEM then the epoch is NULL, but can fix it in following patch. thx

Reply:
"*sync_args->epoch" will be NULL under -DER_NOMEM case, but the "sync_args->epoch" cannot be NULL anyway.